docs/content/SegmentedControl.mdx

```diff
@@ -0,0 +1,190 @@
+---
+title: SegmentedControl
+status: Draft
+description: Used to select an option from a short list and immediately apply the selection
+---
+
+## Examples
+
+### Simple
+
+```jsx live drafts
+<SegmentedControl aria-label="File view">
+  <SegmentedControl.Item>Preview</SegmentedControl.Item>
+  <SegmentedControl.Item>Raw</SegmentedControl.Item>
+  <SegmentedControl.Item>Blame</SegmentedControl.Item>
+</SegmentedControl>
+```
+
+### With icons and labels
+
+```jsx live drafts
+<SegmentedControl aria-label="File view">
+  <SegmentedControl.Item leadingIcon={EyeIcon}>Preview</SegmentedControl.Item>
+  <SegmentedControl.Item leadingIcon={FileCodeIcon}>Raw</SegmentedControl.Item>
+  <SegmentedControl.Item leadingIcon={PeopleIcon}>Blame</SegmentedControl.Item>
+</SegmentedControl>
+```
+
+### With icons only
+
+```jsx live drafts
+<SegmentedControl aria-label="File view">
+  <SegmentedControl.IconItem icon={EyeIcon} aria-label="Preview" />
+  <SegmentedControl.IconItem icon={FileCodeIcon} aria-label="Raw" />
+  <SegmentedControl.IconItem icon={PeopleIcon} aria-label="Blame" />
+</SegmentedControl>
+```
+
+### With labels hidden on smaller viewports
+
+```jsx live drafts
+<SegmentedControl aria-label="File view" hideLabels={{narrow: false, large: true}}>
+  <SegmentedControl.Item leadingIcon={EyeIcon}>Preview</SegmentedControl.Item>
+  <SegmentedControl.Item leadingIcon={FileCodeIcon}>Raw</SegmentedControl.Item>
+  <SegmentedControl.Item leadingIcon={PeopleIcon}>Blame</SegmentedControl.Item>
+</SegmentedControl>
+```
+
+### Convert to a dropdown on smaller viewports
+
+```jsx live drafts
+<SegmentedControl aria-label="File view" asDropdown={{narrow: true, large: false}}>
+  <SegmentedControl.Item leadingIcon={EyeIcon}>Preview</SegmentedControl.Item>
+  <SegmentedControl.Item leadingIcon={FileCodeIcon}>Raw</SegmentedControl.Item>
+  <SegmentedControl.Item leadingIcon={PeopleIcon}>Blame</SegmentedControl.Item>
+</SegmentedControl>
+```
+
+### Fill width of parent
+
+```jsx live drafts
+<SegmentedControl block aria-label="File view">
```

Comment:
Do we have an example of this already? This seems to come up often, would be nice to have the same prop for it - block / fullWidth

I think in Button, we went in favor of sx={{width: '100%'}} instead of `fullWidth`, I don't remember the exact reason though cc @pksjce

Reply:
I went with `block` because that's what I've seen on other components. I'm also fine with `sx={{width: '100%'}}`

Reply:
I like it, I'm team block :D

Reply:
During our engineering sync, we decided to move forward with a `fullWidth` prop. @mperrotti will write a brief ADR for this decision.